src/app/api/query/base.test.ts

```diff
@@ -0,0 +1,61 @@
+import * as reactQuery from "@tanstack/react-query";
+
+import { useWebsocketAwareQuery } from "./base";
+
+import { rootState, statusState } from "@/testing/factories";
+import { renderHookWithMockStore } from "@/testing/utils";
+
+vi.mock("@tanstack/react-query");
+
+const mockQueryFn = vi.fn();
+const mockQueryKey = ["zones"] as const;
+
+beforeEach(() => {
+  vi.resetAllMocks();
+  const mockQueryClient = {
+    invalidateQueries: vi.fn(),
+  } as unknown as reactQuery.QueryClient;
```

Comment:
I don't know how I feel about this double type cast. Again, if this is the cleanest way it can be done then I'm happy as well, but ideally there would be a better way of doing this